labs-ios-starter/Surveys/View Controllers/TopicQuestionsViewController.swift

```diff
@@ -168,14 +161,92 @@ extension TopicQuestionsViewController: UICollectionViewDataSource, UICollection
     }
 }
 
-// MARK: - Segmented Control Delegate -
+extension TopicQuestionsViewController: UIPickerViewDelegate, UIPickerViewDataSource {
+    func numberOfComponents(in pickerView: UIPickerView) -> Int {
+        1
+    }
+
+    func pickerView(_ pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
+        questions?.count ?? 0
+    }
+
+    func pickerView(_ pickerView: UIPickerView, viewForRow row: Int, forComponent component: Int, reusing view: UIView?) -> UIView {
+        guard let questions = questions else {
+            print("questions array empty")
+            return UIView()
+        }
+        // setup label
+        let padding: CGFloat = 20
+        let pickerLabelHeight: CGFloat = 50
+        let pickerLabel = UILabel(frame: CGRect(x: 0,
+                                                y: 0,
+                                                width: pickerView.frame.size.width - padding,
+                                                height: pickerLabelHeight)
+        )
+
+        let font = UIFont(name: "Apple Symbols", size: 24)
+        pickerLabel.numberOfLines = 0
+        pickerLabel.font = font
+        pickerLabel.text = questions[row].question
+        pickerLabel.sizeToFit()
+        pickerLabel.lineBreakMode = .byWordWrapping
+
+        let stackViewHeight: CGFloat = 70
+        let stackView = UIStackView(frame: CGRect(x: 0,
+                                                  y: 0,
+                                                  width: pickerView.frame.size.width - padding,
+                                                  height: stackViewHeight)
+        )
+
+        stackView.spacing = 0
+        stackView.distribution = .fillProportionally
+        stackView.alignment = .center
+        stackView.axis = .vertical
+        stackView.addArrangedSubview(pickerLabel)
+
+        if row != questions.count - 1 {
+            // down arrow
+            let width: CGFloat = 40
+            let height = width / 2
+            let arrowView = UIButton(frame: CGRect(x: 0,
+                                                   y: 0,
+                                                   width: width,
+                                                   height: height)
+            )
+
+            let image = UIImage(systemName: "chevron.down")
+            arrowView.setImage(image, for: .normal)
+            stackView.addArrangedSubview(arrowView)
+        }
+        return stackView
+    }
+
+    func pickerView(_ pickerView: UIPickerView, rowHeightForComponent component: Int) -> CGFloat {
+        100
+    }
+}
+
+// Move the spinner to the next row when the row is tapped
+extension TopicQuestionsViewController: SingleRowSpinnerDelegate {
+    func updateSpinner() {
+        let row = pickerView.selectedRow(inComponent: 0)
+        if row != pickerView.numberOfRows(inComponent: 0) - 1 { // -1 to account for non 0 based count
+            let nextRow = pickerView.selectedRow(inComponent: 0) + 1
```

Comment:
What stops this from going out of range on the last row?

Reply:
Should you use some kind of modulus logic to "wrap-around" back to the first row? It's hard to tell just by looking at lines of code, but it looks like this would get stuck on the last row.

Reply:
good idea to wrap-around. We can just use an else since we're checking to see if it's not the last row here